src/main/java/space/obminyashka/items_exchange/dao/UserRepository.java

```diff
@@ -52,4 +52,9 @@ public interface UserRepository extends JpaRepository<User, UUID> {
     @Modifying
     @Query("update User u set u.oauth2Login = true where u.email = :email and u.oauth2Login is null")
     void setOAuth2LoginToUserByEmail(String email);
+
+    @Transactional
+    @Modifying
+    @Query("update User u set u.validatedEmail = true where u.username = (select user.username from email_confirmation_code where id=:id)")
```

Comment:
At least that query has to be split into a few lines because it has more than 120 symbols, but don't rush doing that because I have some other concerns about the quantity of the requests in the service.